irr/src/OpenGL/Driver.cpp

```diff
@@ -710,58 +710,43 @@ void COpenGL3DriverBase::drawVertexPrimitiveList(const void *vertices, u32 verte
 
 	setRenderStates3DMode();
 
-	auto &vTypeDesc = getVertexTypeDescription(vType);
-	beginDraw(vTypeDesc, reinterpret_cast<uintptr_t>(vertices));
-	GLenum indexSize = 0;
-
-	switch (iType) {
-	case (EIT_16BIT): {
-		indexSize = GL_UNSIGNED_SHORT;
-		break;
-	}
-	case (EIT_32BIT): {
-		indexSize = GL_UNSIGNED_INT;
-		break;
-	}
-	}
-
-	switch (pType) {
-	case scene::EPT_POINTS:
-	case scene::EPT_POINT_SPRITES:
-		GL.DrawArrays(GL_POINTS, 0, primitiveCount);
-		break;
-	case scene::EPT_LINE_STRIP:
-		GL.DrawElements(GL_LINE_STRIP, primitiveCount + 1, indexSize, indexList);
-		break;
-	case scene::EPT_LINE_LOOP:
-		GL.DrawElements(GL_LINE_LOOP, primitiveCount, indexSize, indexList);
-		break;
-	case scene::EPT_LINES:
-		GL.DrawElements(GL_LINES, primitiveCount * 2, indexSize, indexList);
-		break;
-	case scene::EPT_TRIANGLE_STRIP:
-		GL.DrawElements(GL_TRIANGLE_STRIP, primitiveCount + 2, indexSize, indexList);
-		break;
-	case scene::EPT_TRIANGLE_FAN:
-		GL.DrawElements(GL_TRIANGLE_FAN, primitiveCount + 2, indexSize, indexList);
-		break;
-	case scene::EPT_TRIANGLES:
-		GL.DrawElements((LastMaterial.Wireframe) ? GL_LINES : (LastMaterial.PointCloud) ? GL_POINTS
-																						: GL_TRIANGLES,
-				primitiveCount * 3, indexSize, indexList);
-		break;
-	default:
-		break;
-	}
-
-	endDraw(vTypeDesc);
+	renderArray(vertices, indexList, primitiveCount, vType, pType, iType);
 }
 
+//! draws a vertex primitive list in 2d
 void COpenGL3DriverBase::draw2DVertexPrimitiveList(const void *vertices, u32 vertexCount,
 		const void *indexList, u32 primitiveCount,
 		E_VERTEX_TYPE vType, scene::E_PRIMITIVE_TYPE pType, E_INDEX_TYPE iType)
 {
-	os::Printer::log("draw2DVertexPrimitiveList unimplemented", ELL_ERROR);
+	if (!primitiveCount || !vertexCount)
+		return;
+
+	if (!vertices)
+		return;
+
+	if (!checkPrimitiveCount(primitiveCount))
+		return;
+
+	CNullDriver::draw2DVertexPrimitiveList(vertices, vertexCount, indexList, primitiveCount, vType, pType, iType);
+
+	chooseMaterial2D();
```

Comment:
also this call needs to be removed, since it will always override the material the caller may have set before.